File: utils/network.js
const axios = require('axios').default
const logger = require('./logger')
const user = require('../credentials')
const COMMON_HEADER = [
  'Content-Type: application/json',,
  'X-Unity-Version: 2017.4.39f1',,
  'User-Agent: Dalvik/2.1.0 (Linux; U; Android 6.0.1; X Build/V417IR)',,
  'Connection: Keep-Alive',
]
const PASSPORT_HEADER = [
  'Content-Type: application/x-www-form-urlencoded; charset=UTF-8',,
  'User-Agent: Dalvik/2.1.0 (Linux; U; Android 6.0.1; X Build/V417IR)',,
  'Connection: Keep-Alive',
]

// 基础 get/post
// 均自带 commonHeader，使用 Object.assign() 合并 headers
// BTW, Fuck Promise.
function get(url, params = {}, headersAddition = {}) {
  console.log('hello');
  return axios.get(url, {
    headers: Object.assign(COMMON_HEADER, headersAddition),
    params
  })
    .catch((err) => {
      logger.halt(err)
    })
}
function post(url, data = {}, headersAddition = {}) {
  logger.out('POST ' + url)
  return axios.post(url, {
    headers: Object.assign(COMMON_HEADER, headersAddition),
    data
  })
    .catch((err) => {
      logger.halt(err)
    })
}
  
module.exports = {

  // 命名规律：请求方法+服务器

  // 获取 配置信息
  getConfig: (uri) => get(user.server.CONF + uri),

  // 访问 账号认证服务器
  postAuth: (uri, data = {}) => post(user.server.AUTH + uri, data),

  // 访问 游戏服务器
  postGame(uri, data = {}, player) {
    let headers = {
      uid: player.uid,
      secret: player.secret,
      seqnum: player.seqnum
    }
    return post(user.server.GAME + uri, data, headers)
  },

  // 访问 Yostar 账号服务器（外服专用）
  postPassport: (uri, data = {}) => post(user.server.PASSPORT + uri, data, PASSPORT_HEADER)

}

File: utils/network.py
import urllib3
from . import logger
from ..credentials import user

COMMON_HEADER = {
  'Content-Type': 'application/json',
  'X-Unity-Version': '2017.4.39f1',
  'User-Agent': 'Dalvik/2.1.0 (Linux; U; Android 6.0.1; X Build/V417IR)',
  'Connection': 'Keep-Alive'
}
PASSPORT_HEADER = {
  'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
  'User-Agent': 'Dalvik/2.1.0 (Linux; U; Android 6.0.1; X Build/V417IR)',
  'Connection': 'Keep-Alive'
}

http = urllib3.PoolManager()

# 基础请求方法
# 均自带 COMMON_HEADER 并自动合并头
# 同步写起来有点爽 233

def __get(url, params={}, headersAddition={}, debug=True):
  if debug:
    logger.out('GET ', url)
    logger.out(params)
  return http.request('GET', url, 
    headers=(COMMON_HEADER.items() + headersAddition.items()),
    fields=params)

def __post(url, data={}, headersAddition={}, debug=True):
  if debug:
    logger.out('POST', url)
    logger.out(data)
  return http.request('POST', url,
    headers=(COMMON_HEADER.items() + headersAddition.items()),
    fields=data)

# 获取配置信息
def getConfig(uri):
  return __get(user['server']['conf'] + uri)

# 访问账号认证服务器
def postAuth(uri, data={}):
  return __post(user['server']['auth'] + uri, data)

# 访问游戏服务器
def postGame(uri, player, data={}):
  return __post(user['server']['game'] + uri, data, {
    'uid': player.uid,
    'secret': player.secret,
    'seqnum': player.seqnum
  })

# 访问 Yostar 账号服务器
# 外服专用
def postPassport(uri, data={}):
  return __post(user['server']['passport'] + uri, data, PASSPORT_HEADER)
